src/fabric.c

```diff
@@ -393,7 +393,10 @@ void fi_ini(void)
 	fi_register_provider(VERBS_INIT, NULL);
         /* Initialize the sockets provider last.  This will result in
            it being the least preferred provider. */
-	fi_register_provider(SOCKETS_INIT, NULL);
+        fi_register_provider(SOCKETS_INIT, NULL);
+#ifdef HAVE_MXM
+        fi_register_provider(MXM_INIT, NULL);
+#endif
```

Comment:
The sockets provider needs to be listed last.  See the comment right above this change.  There should also be no need to have an ifdef in the code.  Please see how the other providers handle this.